shims/spark300/src/main/scala/com/nvidia/spark/rapids/shims/spark300/Spark300Shims.scala

```diff
@@ -436,4 +445,68 @@ class Spark300Shims extends SparkShims {
       explicitMetadata: Option[Metadata]): Alias = {
     Alias(child, name)(exprId, qualifier, explicitMetadata)
   }
+
+  override def getLocationWithAlluxioReplace(
+      conf: RapidsConf,
+      relation: HadoopFsRelation,
+      partitionFilters: Seq[Expression],
+      dataFilters: Seq[Expression]): FileIndex = {
+
+    val alluxioFilesReplace: Option[Seq[String]] = conf.getAlluxioFilesReplace
+
+    if (alluxioFilesReplace.isDefined) {
+      // alluxioFilesReplace: Seq("key->value", "key1->value1")
+      val replaceMap = new mutable.HashMap[String, String]()
+      alluxioFilesReplace.get.foreach(rule => {
+        val ruleSplit = rule.split("->")
+        if (ruleSplit.size == 2) {
+          replaceMap += ruleSplit(0).trim -> ruleSplit(1).trim
+        } else {
+          logWarning("Wrong set for spark.rapids.alluxio.pathsToReplace " + rule)
+        }
+      })
+
+      if (replaceMap.size > 0) {
+        val listFiles: Seq[PartitionDirectory] = relation.location.listFiles(
+          partitionFilters, dataFilters)
+
+        val replaceFunc = (f: Path) => {
+          val matchedSet = replaceMap.keySet.filter(reg => f.toString.startsWith(reg))
+          if (matchedSet.size > 0) {
+            new Path(
+              f.toString.replaceFirst(matchedSet.head, replaceMap(matchedSet.head)))
+          } else {
+            f
+          }
+        }
+
+        val inputFiles: Seq[Path] = listFiles.flatMap(partitionDir => {
+          partitionDir.files.map(f => replaceFunc(f.getPath))
+        }).toSet.toSeq
```

Comment:
why do we do toSet.ToSeq?  it should be a Seq already.

Reply:
why do we have replications of same file? was this copied from somewhere in Spark?  what if user specified a directory twice because they wanted to read it twice?

Reply:
Done